src/components/Composer/index.js

```diff
@@ -63,6 +63,9 @@ export class Composer extends Component {
 	}
 
 	handleInput = (onChange) => () => {
+		if (inputLock) {
```

Comment:
This is not a state, the variable is not declared as a component state property.

Reply:
@renatobecker 
Does my latest commit fulfill the requirement?
If not, can you provide a bit hint on that?
The build and behavior seems alright.

Reply:
@renatobecker
I dont know how to implement a state variable as I have to use setState() in onCompositionStart() and onCompositionEnd(),
which is part of the render function. I read about not using setState() in render function but I dont see a way to avoid it.
[do not use setState() in render](https://stackoverflow.com/a/51677449),
I currently call the function handleInputLock to setState but I doubt it is keeping render function pure.
Any suggestion or hint?